packages/consensus/source/round-state.ts

```diff
@@ -119,6 +120,21 @@ export class RoundState implements Contracts.Consensus.RoundState {
 		throw new Error("Block is not available, because proposal is not set or deserialized");
 	}
 
+	public consumeGas(amount: number): void {
```

Comment:
Extract this part into new class BlockProcessorResult implemented in block processor. Consensus shouldn't be aware that gas exists. This belongs under tx/block domain.

We should replace boolean value #processorResult with more complex class that have additional fields and methods:

pass (block is valid)
totalGasUsed
...
Gas verification logic can be implemented directly on processor.